src/dataflow/components/dataflow-program.tsx

```diff
@@ -220,6 +223,36 @@ export class DataflowProgram extends BaseComponent<IProps, IState> {
   private setProgramRunTime = (time: number) => {
     this.setState({programRunTime: time});
   }
+  private generateProgramData = () => {
+    let programName: any = "my-program";
+    const endTimePad = 24 * 60 * 60 * 1000;
```

Comment:
end time should be specified before a program can be run - this should come out when we have that.

Reply:
Ah, we have that in the UI too.